### PR TITLE
Add more test conditions for chem kinetics

### PR DESCRIPTION
**Pull request by tulioricci**
- Split `test_eos` and move chemical kinetics tests to `test_chemistry`
- Add extra chemistry verification at different pressure conditions
- Add verification of species entropy and Gibbs energy (used under the hood in reversible reactions)
- Add lewis=1.0 test option to `test_transport`
-  Explicitly export the mechanism and read it back for better assessment of test coverage in `test_eos`, `test_chemistry` and `test_transport`
- Tried to remove deprecation warns from Pyrometheus but left as a comment instead. Need https://github.com/illinois-ceesd/drivers_y3-prediction/pull/53

**Questions for the review**:
- [ ] Is the scope and purpose of the PR clear?
  - [ ] The PR should have a description.
  - [ ] The PR should have a guide if needed (e.g., an ordering).
- [ ] Is every top-level method and class documented? Are things that should be documented actually so?
- [ ] Is the interface understandable? (I.e. can someone figure out what stuff does?) Is it well-defined?
- [ ] Does the implementation do what the docstring claims?
- [ ] Is everything that is implemented covered by tests?
- [ ] Do you see any immediate risks or performance disadvantages with the design? Example: what do interface normals attach to?

### CODE REVIEW

File: test/test_eos.py [hunk 1]
```diff
@@ -328,7 +339,7 @@ def inf_norm(x):
             assert inf_norm(conc[spec]) < 1e-14
 
 
-# TODO remove this test.. It is already covered in the other ones
+# FIXME This test is kinda redudant
```
Comment (MTCam) [inline test/test_eos.py:342]: This comment goes from bad to worse.  Can you make it a little more clear to dummies?

Comment (tulioricci) [reply test/test_eos.py:342]: Essentially, we have 3 tests:
1) pyrometheus vs cantera
2) mirge-com vs pyrometheus
3) mirge-com vs cantera

My original suggestion was to drop this out. But keeping it is also fine. So either I remove the comment or the test.

Comment (MTCam) [reply test/test_eos.py:342]: I would just leave the tests, then.  It is not entirely clear there is real redundancy here without more details.   I can see motivation for making sure that our massage of Pyro into mirgecom didn't break pyro, that mirgecom gets the same answers as pyro when using mirgecom API, and that mirgecom matches Cantera when we do a dynamic problem.

Comment (tulioricci) [reply test/test_eos.py]: :+1:

**Review by MTCam — APPROVED** [review]

**Pull request merged**